src/primitives/sinsemilla.rs

```diff
@@ -1,19 +1,37 @@
 //! The Sinsemilla hash function and commitment scheme.
 
-use group::Wnaf;
+use group::{Curve, Wnaf};
 use halo2::arithmetic::{CurveAffine, CurveExt};
 use pasta_curves::pallas;
 use subtle::CtOption;
 
-use crate::spec::{extract_p_bottom, i2lebsp};
-
 mod addition;
 use self::addition::IncompletePoint;
-
-mod constants;
 mod sinsemilla_s;
-pub use constants::*;
-pub(crate) use sinsemilla_s::*;
+pub use sinsemilla_s::SINSEMILLA_S;
+
+/// Number of bits of each message piece in $\mathsf{SinsemillaHashToPoint}$
+pub const K: usize = 10;
+
+/// $\frac{1}{2^K}$
+pub const INV_TWO_POW_K: [u8; 32] = [
+    1, 0, 192, 196, 160, 229, 70, 82, 221, 165, 74, 202, 85, 7, 62, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 240, 63,
+];
+
+/// The largest integer such that $2^c \leq (r_P - 1) / 2$, where $r_P$ is the order
+/// of Pallas.
+pub const C: usize = 253;
```

Comment:
Why are these duplicated in `constants/sinsemilla.rs` and here?

Note that `K` will need to be generic for other uses of Sinsemilla than Orchard.

Reply:
Without `const_evaluatable_checked` it's hard for `K` to be generic. For now we'll leave it as a requirement of the chip implementation, and we can figure out how to refactor it later.

Reply:
(as for the duplicated constants, we can clean those up later as well).